nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -217,30 +175,6 @@ sub GrubFs {
             }
             $search .= $matches[0];
         }
-
-        # BTRFS is a special case in that we need to fix the referenced path based on subvolumes
-        if ($fs->type eq 'btrfs') {
-            my ($status, @id_info) = runCommand("@btrfsprogs@/bin/btrfs", "subvol", "show", @{[$fs->mount]});
-            if ($status != 0) {
-                die "Failed to retrieve subvolume info for @{[$fs->mount]}\n";
-            }
-            my @ids = join("\n", @id_info) =~ m/^(?!\/\n).*Subvolume ID:[ \t\n]*([0-9]+)/s;
-            if ($#ids > 0) {
-                die "Btrfs subvol name for @{[$fs->device]} listed multiple times in mount\n"
-            } elsif ($#ids == 0) {
-                my ($status, @path_info) = runCommand("@btrfsprogs@/bin/btrfs", "subvol", "list", @{[$fs->mount]});
-                if ($status != 0) {
-                    die "Failed to find @{[$fs->mount]} subvolume id from btrfs\n";
-                }
-                my @paths = join("", @path_info) =~ m/ID $ids[0] [^\n]* path ([^\n]*)/;
-                if ($#paths > 0) {
-                    die "Btrfs returned multiple paths for a single subvolume id, mountpoint @{[$fs->mount]}\n";
-                } elsif ($#paths != 0) {
-                    die "Btrfs did not return a path for the subvolume at @{[$fs->mount]}\n";
-                }
-                $path = "/$paths[0]$path";
-            }
-        }
```

Comment:
Btrfs no longer needs to be treated specially here since `$fs->root` will contain the subvolume path and will be resolved the same way bind mounts are resolved.